### ISSUE TITLE
Suggestion: using a mask to do avoid_pad in losses.py

### ISSUE THREAD

**Issue by jjery2243542** [speechbrain/speechbrain#53]
Now the implementation is iterating through each sentence and finding the actual length for each one.
Using a mask to do it could be more elegant and efficient. 

example:
mask = length_to_mask(lengths, max_len=target.shape[1])
loss = cost(prob, lab) # without reduction
loss = torch.sum(loss * mask) / torch.sum(mask)
